syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -39,49 +34,46 @@ func newGoBinaryPackage(name, version, h1Digest, goVersion, architecture string,
 	return p
 }
 
-func parseGoBin(location source.Location, reader io.ReadCloser, opener exeOpener) (pkgs []pkg.Package, err error) {
-	var exes []exe
-	// it has been found that there are stdlib paths within openExe that can panic. We want to prevent this behavior
-	// bubbling up and halting execution. For this reason we try to recover from any panic and return an error.
-	defer func() {
-		if r := recover(); r != nil {
-			err = fmt.Errorf("recovered from panic while parse go binary at %q: %+v", location.RealPath, r)
+func getGOARCH(settings []debug.BuildSetting) string {
```

Comment:
this had been based on the architecture from the binary --it's ok to now source this from the build settings, however, do we know if this value is guaranteed to be set for all modules? (honest question, I haven't checked the go source code on this one)

Reply:
https://github.com/golang/go/blob/bf97c99b62fe7d6652cc8c807dbc91998d488a01/src/cmd/go/internal/load/pkg.go#L2336 sets the architecture, it happens inside `setBuildInfo()` which states:

```
// setBuildInfo gathers build information, formats it as a string to be
// embedded in the binary, then sets p.Internal.BuildInfo to that string.
// setBuildInfo should only be called on a main package with no errors.
//
// This information can be retrieved using debug.ReadBuildInfo.
```
https://github.com/golang/go/blob/bf97c99b62fe7d6652cc8c807dbc91998d488a01/src/cmd/go/internal/load/pkg.go#L2211L2218

Reply:
I am looking into compatibility with older version of go binaries that won't have the key/pair

Reply:
> I am looking into compatibility with older version of go binaries that won't have the key/pair

I think this hints at still sourcing this from the binary instead of the build settings.